crates/cairo-profiler/src/trace_reader/function_trace_builder.rs

```diff
@@ -208,3 +191,28 @@ fn maybe_sierra_statement_index_by_pc(
         MaybeSierraStatementIndex::SierraStatementIndex(statement_index)
     }
 }
+
+fn build_current_call_stack(
+    call_stack: &CallStack,
+    current_function_name: FunctionName,
+    show_inlined_functions: bool,
+    sierra_statement_idx: StatementIdx,
+    statements_functions_map: Option<&StatementsFunctionsMap>,
+) -> VecWithLimitedCapacity<FunctionCall> {
+    let mut current_call_stack = call_stack.current_call_stack().clone();
+    current_call_stack.push(FunctionCall::InternalFunctionCall(
+        InternalFunctionCall::NonInlined(current_function_name),
+    ));
+
+    if show_inlined_functions {
```

Comment:
Does it make sense to also check if there is any free space in the `current_call_stack`? If it already contains `max_capacity` elements, we won't be able to add anything else.

Reply:
I think I can actually pass `VecWithCapacity` to `build_original_call_stack_with_inlined_calls` and it would solve this weird vector deconstruction (and truncating the vector)